pkg/migrations/sql/idempotent/base.sql

```diff
@@ -440,51 +447,70 @@ BEGIN
     --lock as for ALTER TABLE because we are in effect changing the schema here
     --also makes sure the next_position below is correct in terms of concurrency
     EXECUTE format('LOCK TABLE SCHEMA_DATA_SERIES.%I IN SHARE UPDATE EXCLUSIVE MODE', metric_table);
-    --second check after lock
+
     SELECT
-        pos
+        max(pos) + 1
     FROM
         SCHEMA_CATALOG.label_key_position lkp
     WHERE
         lkp.metric_name = get_new_pos_for_key.metric_name
-        AND lkp.key =  get_new_pos_for_key.key_name INTO position;
+    INTO max_position;
 
-    IF FOUND THEN
-        RETURN position;
+    IF max_position IS NULL THEN
+        max_position := 2; -- element 1 reserved for __name__
```

Comment:
Why would a position of 2 be wrong that's the first position for a non `__name__` label. There is no guarantee that a __name__ label will be created first (only that it will have position=1 once it is created). So this is set up to handle the case that there is no entry in the label_key_position table for the metric and we are creating a key that is not __name__ first.